lib/providers/actionProviderComposer.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * @name actionProviderComposer
+ * @description a function that satisfies the Atom IDE CodeActionProvider
+ * interface, running each diagnostic through a series of providers, each
+ * of which may return either a valid CodeAction (optionally wrapped in a
+ * Promise) as defined by Atom IDE or null if a given provider chooses not
+ * to provide an action for the diagnostic
+ * @param {JavaLanguageClient} languageClient the wrapping Java language client
+ * @param {[function]} providers a list of providers to register
+ * @returns {atom$CodeActionProvider}
+**/
```

Comment:
This documentation format is quite different to what we use.  We're not generating any docs from source at this time though so not sure it matters too much right now.

Reply:
Yeah, I didn't see any to go off of so I just used what I normally use. Would be happy to change or omit altogether if there's a preferred standard.